scripts/compress_images.py

```diff
@@ -0,0 +1,246 @@
+                str(file_path),
+                '-strip',
+                '-compress', compression_type,
+                str(output_file_path)
+            ]
+
+            result = subprocess.run(
+                cmd, capture_output=True, text=True, check=True
+            )
+            if result.returncode == 0 and output_file_path.exists():
+                original_size = file_path.stat().st_size
+                new_size = output_file_path.stat().st_size
+
+                if new_size < original_size * self.tolerance:
+                    image_compressibility_dicts.append({
+                        'path': file_path,
+                        'original_size': original_size,
+                        'new_size': new_size
+                    })
+            else:
+                logging.error(
+                    '[ERROR]: %s occurred on file %s',
+                        result.stderr,
+                        file_path
+                )
+
+        return image_compressibility_dicts
+
+    def find_compressible_images(self) -> List[CompressedImageInfo]:
+        """Find images that can be compressed further.
+
+        Returns:
+            result_images: List[CompressedImageInfo]. List of all compressible
+            images over repository with attributes as path, original_size
+            and new_size.
+        """
+        result_images: List[CompressedImageInfo] = []
+
+        for file_path in self.input_path.glob('**/*.*'):
+            file_extension = file_path.suffix.lower()
+            if file_extension not in self.ALL_IMAGES_EXTENSIONS:
+                continue
+
+            get_tempdir = tempfile.gettempdir()
+            temp_compressed = (
+                pathlib.Path(get_tempdir) / f'compressed_{file_path.name}'
+            )
+            result = self.run_compression_logic(
+                file_path,
+                temp_compressed,
+            )
+            if result:
+                result_images.extend(result)
+
+        return result_images
+
+    def compress_images(
+            self, result_images: List[CompressedImageInfo]
+        ) -> None:
+        """Compress the identified images.
+
+        Args:
+            result_images: List(CompressedImageInfo). List of images
+                to be compressed.
+        """
+        # Remove existing output directory and create a new one.
+        shutil.rmtree(self.output_dir, ignore_errors=True)
+
+        os.makedirs(self.output_dir, exist_ok=True)
+
+        for image in result_images:
+            file_path = image['path']
+
+            # Maintain original directory structure.
+            rel_path = os.path.relpath(file_path, self.input_path)
+            output_file_path = pathlib.Path(self.output_dir) / rel_path
+
+            # Create necessary directories.
+            os.makedirs(os.path.dirname(output_file_path), exist_ok=True)
+            compress_result = self.run_compression_logic( # pylint: disable=unused-variable
```

Comment:
```suggestion
            self.run_compression_logic(
```

No need to save the return value since you don't use it

Reply:
Hey @U8NWXD I made this change on your suggestions and reported you even about it's non-used behaivour, PTAL here https://github.com/oppia/oppia/pull/21694#discussion_r2008042508

Reply:
Yes, I previously asked that we return the file size info from `run_compression_logic()`, and I still think that's the right approach. To clarify, I'm not suggesting that you change what `run_compression_logic()` returns; instead, I'm suggesting that since we don't need the return value from this particular invocation of `run_compression_logic`, you don't need to save the return value in `compress_result`. You can just call `self.run_compression_logic(...)` and discard the return value. Does that make sense?

Reply:
Ahh, you're saying in that way. That's totally makes sense to me, and fixed it. Sorry for the delay.